src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumerIntegrationErrorModeTest.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.listener.ContainerProperties;
+import org.springframework.kafka.listener.KafkaMessageListenerContainer;
+import org.springframework.kafka.listener.MessageListener;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.ContainerTestUtils;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.util.concurrent.ListenableFuture;
+import uk.gov.companieshouse.kafka.consumer.resilience.CHConsumerType;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@DirtiesContext
+@EmbeddedKafka
+@TestPropertySource(properties="uk.gov.companieshouse.item-handler.error-consumer=true")
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class OrdersKafkaConsumerIntegrationErrorModeTest {
+    private static final String ORDER_RECEIVED_TOPIC = "order-received";
+    private static final String ORDER_RECEIVED_TOPIC_RETRY = "order-received-retry";
+    private static final String ORDER_RECEIVED_TOPIC_ERROR = "order-received-error";
+    private static final String CONSUMER_GROUP_MAIN_RETRY = "order-received-main-retry";
+    private static final String CONSUMER_GROUP_MAIN_ERROR = "order-received-error";
+    private static final String ORDER_RECEIVED_URI = "/order/ORDER-12345";
+    @Value("${kafka.broker.addresses}")
+    private String brokerAddresses;
+
+    private KafkaTemplate<String, String> template;
+
+    private KafkaMessageListenerContainer<String, String> container;
+
+    private BlockingQueue<ConsumerRecord<String, String>> records;
+
+    @Autowired
+    private OrdersKafkaConsumerWrapper consumerWrapper;
+
```

Comment:
The fact that the same `consumerWrapper` is used throughout these tests makes the code reader whether the order URI could be cached from test to test for example?

Reply:
Updated.